### PR TITLE
Refactor test types

### PR DESCRIPTION
**Pull request by piontec**
This PR:
-  simplfies the code by:
  - removing almost fully redundant StepType and TestType data types
  - using base app operations (`create`, `delete`) from `pytest-helm-charts`
- adds tests for main part of pytest runner

### CODE REVIEW

**Review by ubergesundheit — APPROVED** [review]
LGTM from a quick glance, although I'll probably will look at the whole codebase more once we'll be getting to circleCI integration

File: tests/pytest/test_pytest_runner.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            "--cluster-type",
+            "mock",
+            "--kube-config",
+            mock_kube_config_path,
+            "--chart-path",
+            config.chart_file,
+            "--chart-version",
+            mock_app_version,
+            "--chart-extra-info",
+            "external_cluster_version=1.19.1",
```
Comment (rossf7) [inline tests/pytest/test_pytest_runner.py:101]: `1.19.1` is the kind version?

What about having a constant for this?  So it's easier to find and update.

Also 1.19.1 is quite old now. For the Go tests we use 1.21.1 and the latest kind tag is 1.21.2. I think we should use at least K8s 1.20.2.

https://quay.io/repository/giantswarm/kind-node

**Review by rossf7 — COMMENTED** [review]
My review is a bit shallow due to a lack of python knowledge but I like the restructuring. 👍 

Just a query on the k8s / kind version.

File: tests/pytest/test_pytest_runner.py (hunk 1 above)
Comment (piontec) [reply tests/pytest/test_pytest_runner.py:101]: It's total mock, doesn't have anything to do with real version used anywhere. Real cluster manager is not used here, only the mocked one, so this value is never used anywhere :) Maybe I should set it to something ridiculous, so it's obvious :)

Comment (rossf7) [reply tests/pytest/test_pytest_runner.py:101]: OK I missed that completely. 😹  What about `2.0.0`? :trollface:  

I think it's fine maybe just bump it to 1.21. It was the 1.19 that triggered me.

Comment (rossf7) [reply tests/pytest/test_pytest_runner.py:101]: OK its fine as it. No need to waste more time on this.

**Pull request merged**